sound/hda/intel-sdw-acpi.c

```diff
@@ -79,10 +78,21 @@ sdw_intel_scan_controller(struct sdw_intel_acpi_info *info)
 	 * We will check the hardware capabilities in the startup() step
 	 */
 
-	if (ret) {
-		dev_err(&adev->dev,
-			"Failed to read mipi-sdw-master-count: %d\n", ret);
-		return -EINVAL;
+	count = 0;
+	ret = fwnode_property_read_u32(acpi_fwnode_handle(adev),
+				       "mipi-sdw-manager-list", &list);
+	if (ret < 0) {
+		ret = fwnode_property_read_u32(acpi_fwnode_handle(adev),
+					       "mipi-sdw-master-count", &count);
+		if (ret) {
+			dev_err(&adev->dev,
+				"Failed to read mipi-sdw-manager-list or mipi-sdw-master-count: %d\n", ret);
+			return -EINVAL;
+		}
+
+		list = GENMASK(count - 1, 0);
```

Comment:
Is it possible that count = 0?

Reply:
Good catch, that would be a problem, I should test this indeed for avoid a nonsensical loop.

Reply:
That's true, we already trap the count = 0. Thanks @ranj063 for the comment.

https://github.com/thesofproject/linux/pull/4857/files/8d44869d3c076940fd95ca751894916b4eae49e6..a3edd2e5dde37524d4f5f103b7c1a6974c7eba02#diff-55d16501c077af0123a7127463a80e233d5cace32c9acf2f26d3289348a9602dR105

So while it's a bit odd to create an all-1 link value, there isn't any bad consequences and the behavior remains the same as before.